src/fontra/views/editor/visualization-layer-definitions.js

```diff
@@ -171,6 +171,66 @@ registerVisualizationLayerDefinition({
   },
 });
 
+registerVisualizationLayerDefinition({
+  identifier: "fontra.verticalMetrics",
+  name: "Vertical font metrics",
+  selectionMode: "editing",
+  userSwitchable: true,
+  defaultOn: true,
+  zIndex: 100,
+  screenParameters: { strokeWidth: 1 },
+  colors: { strokeColor: "#0004", zoneColor: "#0001" },
+  colorsDarkMode: { strokeColor: "#FFF6", zoneColor: "#FFF1" },
+  draw: (context, positionedGlyph, parameters, model, controller) => {
+    context.strokeStyle = parameters.strokeColor;
+    context.lineWidth = parameters.strokeWidth;
+
+    if (model.fontSourceInstance === undefined) {
+      return;
+    }
+    const verticalMetrics = model.fontSourceInstance.verticalMetrics;
+    // const italicAngle = model.fontSourceInstance.italicAngle
+    //   ? model.fontSourceInstance.italicAngle * -1
+    //   : 0;
+    const glyphWidth = positionedGlyph.glyph.xAdvance
+      ? positionedGlyph.glyph.xAdvance
+      : 0;
+
+    // glyph box
+    let pathBox = new Path2D();
+    pathBox.rect(
+      0,
+      verticalMetrics.descender.value,
+      positionedGlyph.glyph.xAdvance,
+      verticalMetrics.ascender.value - verticalMetrics.descender.value
+    );
+
+    // collect paths: vertical metrics and alignment zones
+    let pathZones = new Path2D();
+    for (const [key, metric] of Object.entries(verticalMetrics)) {
+      let pathZone = new Path2D();
+      pathZone.rect(0, metric.value, glyphWidth, metric.zone);
+      pathZones.addPath(pathZone);
+
+      let pathMetric = new Path2D();
+      pathMetric.moveTo(0, metric.value);
+      pathMetric.lineTo(glyphWidth, metric.value);
+      pathBox.addPath(pathMetric);
+    }
+
+    // draw zones (with filled path)
+    let path2DZones = new Path2D();
+    path2DZones.addPath(pathZones); //, new DOMMatrix().skewX(italicAngle));
```

Comment:
Why the additional path object? Same below for `path2DBox`.

Reply:
This is/was related to the 'italicAngle' I commented out for now. I needed it to add `new DOMMetrix` with skewX.

Reply:
Ah, sorry. I misunderstood your question. the second additional is required, because the one has a fill (zones) and the other is stroke only. Is there a way to combine both in one?

Reply:
I think both are redundant because of the removal of the skew transform. A new Path2D is made and one path is added to it without change. Maybe best to remove the italicAngle code after all, it will be trivial enough to add back in later if needed.

Btw. I think _all_ the `let`s should be `const` in this function. I don't see a single case where a `const` wouldn't work. `const` should always be the default.

Reply:
OK, I removed the italicAngle code and used const instead of let.